src/main/java/io/cdap/plugin/http/sink/batch/HTTPRecordWriter.java

```diff
@@ -350,30 +363,38 @@ private void flushMessageBuffer() {
     contentType = messageBuffer.getContentType();
     try {
       Awaitility
-        .await().with()
-        .pollInterval(pollInterval)
-        .pollDelay(config.getWaitTimeBetweenPages(), TimeUnit.MILLISECONDS)
-        .timeout(config.getMaxRetryDuration(), TimeUnit.SECONDS)
-        .until(this::executeHTTPServiceAndCheckStatusCode);
+              .await().with()
+              .pollInterval(pollInterval)
+              .pollDelay(config.getWaitTimeBetweenPages(), TimeUnit.MILLISECONDS)
+              .timeout(config.getMaxRetryDuration(), TimeUnit.SECONDS)
+              .until(this::executeHTTPServiceAndCheckStatusCode);
     } catch (Exception e) {
-      throw new RuntimeException("Error while executing http request for remaining input messages " +
-                                   "after the batch execution. " + e);
+      String errorMessage = "Error while executing http request for remaining input messages" +
+              " after the batch execution.";
+      throw ErrorUtils.getProgramFailureException(new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN),
+              errorMessage, String.format("Error message: %s", errorMessage), ErrorType.UNKNOWN, true,
```

Comment:
similar comment here

Reply:
Also don't we have status code information here?

Reply:
updated